### PR TITLE
Permission return

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-11904

## Description of Intent of Change(s)
- For cross account principals. Do not return the platform default permissions for them.

- When listing groups for cross account principal, do not return platform group since
they will not inherit the platform default group.

- Only canned roles could be assigned to cross account request principal: when cross account request is created, we verify that the assigned roles are system roles.

- Add a verification in the middleware to check the cross account requests.
Verify if the user is internal and email address ends with @redhat.com.

## Local Testing
Unit tests have covered most cases.

## Checklist
- [ ] if API spec changes are required, is the spec updated?

- [ ] are there any pre/post merge actions required? if so, document here.

- [ ] are theses changes covered by unit tests?

- [ ] if warranted, are documentation changes accounted for?

- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?

- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

File: rbac/management/querysets.py [hunk 1]
```diff
@@ -71,7 +71,9 @@ def get_group_queryset(request):
 
     username = request.query_params.get("username")
     if username:
-        get_principal(username, request.user.account)
+        principal = get_principal(username, request.user.account)
+        if principal.cross_account:
+            return Group.objects.filter(principals__username__iexact=username)
```
Comment (coderbydesign) [inline rbac/management/querysets.py:76]: We just want to return an empty queryset here, correct?

File: rbac/management/utils.py [hunk 2]
```diff
@@ -118,6 +121,8 @@ def groups_for_principal(principal, **kwargs):
         assigned_group_set = assigned_group_set.prefetch_related(prefetch_lookups)
         platform_default_group_set = platform_default_group_set.prefetch_related(prefetch_lookups)
 
+    if principal.cross_account:
+        return set(assigned_group_set)
```
Comment (coderbydesign) [inline rbac/management/utils.py:125]: Same here with regard to returning an empty set?

File: rbac/management/querysets.py (hunk 1 above)
Comment (astrozzc) [reply rbac/management/querysets.py:76]: Oh! Right, thanks for reminding! Almost forgot about that. I will change it

**Review by wcmitchell — APPROVED** [review]
LGTM

**Pull request merged**